crawl-ref/source/makeitem.cc

```diff
@@ -1656,13 +1649,26 @@ static void _generate_misc_item(item_def& item, int force_type, int item_level)
     case MISC_TIN_OF_TREMORSTONES:
     case MISC_CONDENSER_VANE:
     case MISC_GRAVITAMBOURINE:
-        you.generated_misc.insert(typ);
+        you.generated_misc.insert(item.sub_type);
```

Comment:
Shouldn't `item.sub_type` be `static_cast<misc_item_type>(item.sub_type)` here?

Reply:
It should be, yes. Hopefully fixed now (I decided it would be better if this function just took `misc_item_type` instead of the full `item_def` reference.